projects/igniteui-angular/src/lib/select/select.component.ts

```diff
@@ -182,13 +182,21 @@ export class IgxSelectComponent extends IgxDropDownComponent implements IgxSelec
 
     /**
      * An @Input property that sets how the select will be styled.
-     * The allowed values are `line`, `box` and `border`. The default is `line`.
+     * The allowed values are `line`, `box` and `border`. The input-group default is `line`.
      * ```html
      * <igx-select [type]="'box'"></igx-select>
      * ```
      */
     @Input()
-    public type = 'line';
+    public set type(value: IgxInputGroupType) {
```

Comment:
This implementation works fine, but I feel like it should be the same as in the `IgxInputGroup` component. If we expose the input to be templateable or if (for some reason) a user passes items that have and `input` in them, those should behave consistently w/ the input group from the view template. So the select component should get the `InputGroupToken` injected in it's template (optional) and `type` should return `this._type || this._inputGroupToken || 'line'`. As we are passing any specified type to the `input` in the template, there should be no inconsistencies when it is provided.
@damyanpetev , thoughts?

Reply:
I don't think might be an issue, but the input group child may one day not be static and interfere with this setter. So I'd still go that route anyway.